File: contests/hc18/p1/p1_solution.cpp
#include <iostream>
#include <cmath>
#include <algorithm>

int main() {
    int n, mx, my;
    std::cin >> n >> mx >> my;

    long long max_value = 0;
    for (int i = 0; i < n; ++i) {
        int x, y, s;
        std::cin >> x >> y >> s;
        long long distance = std::pow(mx-x,2) + std::pow(my-y,2);
        max_value = std::max(max_value, std::abs(distance-s));
    }
    
    std::cout << max_value << "\n";
}

File: contests/hc18/p1/p1_solution.py
n=int(input())
mx,my=map(int,input().split())
max_value=0
for i in range(n):
    x,y,s=map(int,input().split())
    distance=(mx-x)**2+(my-y)**2
    max_value=max(max_value,abs(distance-s))

print(max_value)